File: backend/app/schemas/trl_validation.py
from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class TRLValidationRunRequest(BaseModel):
    reset_demo_data: bool = False
    scenario_limit: int | None = Field(default=None, ge=1, le=1000)
    validation_mode: Literal["deterministic_regression", "live_llm_staging"] = "deterministic_regression"


class TRLValidationResetResponse(BaseModel):
    deleted_runs: int
    deleted_conversations: int
    deleted_orders: int
    deleted_customers: int = 0


class TRLValidationRunRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    shop_id: UUID
    validation_mode: str
    status: str
    total_scenarios: int
    passed_scenarios: int
    failed_scenarios: int
    metrics_json: dict[str, Any]
    started_at: datetime
    completed_at: datetime | None
    created_by_user_id: UUID | None


class TRLValidationScenarioResultRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    run_id: UUID
    scenario_id: str
    input_json: dict[str, Any]
    expected_json: dict[str, Any]
    actual_json: dict[str, Any]
    passed: bool
    failure_reasons: list[str]
    processing_time_ms: int
    conversation_id: UUID | None
    order_id: UUID | None
    created_at: datetime

File: frontend/src/types/trlValidation.ts
export interface TRLValidationRun {
  id: string;
  shop_id: string;
  status: 'running' | 'completed' | 'failed' | string;
  total_scenarios: number;
  passed_scenarios: number;
  failed_scenarios: number;
  metrics_json: Record<string, unknown>;
  started_at: string;
  completed_at: string | null;
  created_by_user_id: string | null;
}

export interface TRLValidationScenarioResult {
  id: string;
  run_id: string;
  scenario_id: string;
  input_json: Record<string, unknown>;
  expected_json: Record<string, unknown>;
  actual_json: Record<string, unknown>;
  passed: boolean;
  failure_reasons: string[];
  processing_time_ms: number;
  conversation_id: string | null;
  order_id: string | null;
  created_at: string;
}

export interface TRLRiskMetrics {
  invalid_llm_json_count: number;
  safe_fallback_count: number;
  human_handoff_count: number;
  false_positive_order_creation: number;
  false_positive_auto_send: number;
  average_risk_score: number;
  critical_risk_count: number;
  scenario_pass_rate_by_category: Record<string, number>;
  average_processing_latency: number;
  p95_processing_latency: number;
}
